supertokens_python/recipe/emailverification/api/implementation.py

```diff
@@ -59,7 +60,7 @@ async def generate_email_verify_token_post(self, api_options: APIOptions, user_c
         email = await api_options.config.get_email_for_user_id(user_id, user_context)
 
         token_result = await api_options.recipe_implementation.create_email_verification_token(user_id, email, user_context)
-        if token_result.is_email_already_verified:
+        if isinstance(token_result, CreateEmailVerificationTokenEmailAlreadyVerifiedErrorResult):
             return GenerateEmailVerifyTokenPostEmailAlreadyVerifiedErrorResponse()
 
         user = User(user_id, email)
```

Comment:
Below we still have checks for `if token_result.token is None:` -> this should not be needed anymore?

Please make sure to remove these redundant checks. That's partly the whole point of this refactor.